Add buck file for unit test

cccclai opened the pull request:
Summary:
```
buck2 test 'fbcode//mode/dev-nosan' fbcode//executorch/backends/qualcomm/tests/fb:test_qnn_delegate_simulator -- --exact 'executorch/backends/qualcomm/tests/fb:test_qnn_delegate_simulator - test_qnn_backend_mha (executorch.backends.qualcomm.tests.fb.test_qnn_delegate_simulator.TestQNNQuantizedOperatorSimulator)'
```

Differential Revision: D67463640

haowhsu-quic commented on backends/qualcomm/tests/utils.py:
```diff
@@ -131,6 +131,7 @@ class TestQNN(unittest.TestCase):
     use_16a4w: str = "16a4w"
     shared_buffer: bool = False
     enable_x86_64: bool = False
+    dump_intermediate_outputs: bool = False
```
May I learn more about the changes in this file? thank you.

cccclai replied:
It is because this line https://github.com/pytorch/executorch/blob/1d43d91c453dde320186a2952781f1a7b7c650d1/backends/qualcomm/tests/test_qnn_delegate.py#L91

Without setting a default value with `TestQNN`, it fails with
```
AttributeError: type object 'TestQNN' has no attribute 'dump_intermediate_outputs'
```
Following is my code internal for making some customization for the unit test
```
# test_base.py
import os
import tempfile
from typing import Tuple

import numpy as np

import torch

from executorch.backends.qualcomm.tests.test_qnn_delegate import (
    TestQNNQuantizedOperator,
)
from executorch.backends.qualcomm.tests.utils import TestQNN
from executorch.devtools import Inspector
from executorch.examples.qualcomm.utils import generate_inputs
from executorch.exir import ExecutorchProgram, ExecutorchProgramManager
from executorch.extension.pybindings.portable_lib import _load_for_executorch

class TestQNNQuantizedOperatorSimulator(TestQNNQuantizedOperator):
    def setUp(self):
        TestQNN.model = "SM8650"
        TestQNNQuantizedOperator.setUp(self)

    def verify_output(  # noqa: C901
        self,
        module: torch.nn.Module,
        sample_inputs: Tuple[torch.Tensor],
        executorch_prog: ExecutorchProgram | ExecutorchProgramManager,
        etrecord_path: str = "etrecord.bin",
        expected_profile_events: int = -1,
        expected_intermediate_events: int = -1,
        method_index: int = 0,
    ):
        with tempfile.TemporaryDirectory() as tmp_dir:
            (
                input_list,
                ref_outputs,
                pte_fname,
            ) = self._save_model_and_expected_output(
                module,
                executorch_prog.buffer,
                sample_inputs,
                tmp_dir,
            )

            etdump_path = f"{tmp_dir}/etdump.etdp"
            debug_output_path = f"{tmp_dir}/debug_output.bin"

            def validate_profile():
                inspector = Inspector(etdump_path=etdump_path, etrecord=etrecord_path)
                self.assertTrue(
                    len(inspector.to_dataframe().index) == expected_profile_events
                )

            def validate_intermediate_tensor():
                inspector = Inspector(
                    etdump_path=etdump_path, debug_buffer_path=debug_output_path
                )
                for event_block in inspector.event_blocks:
                    if event_block.name == "Execute":
                        self.assertTrue(
                            len(event_block.events) == expected_intermediate_events
                        )

            generate_inputs(tmp_dir, "input_list.txt", [sample_inputs], input_list)
            exec_mod = _load_for_executorch(pte_fname)
            outputs = exec_mod.forward(list(sample_inputs))

            self._assert_outputs_equal(outputs, ref_outputs)

            # Verify the etdump
            if expected_profile_events != -1:
                validate_profile()

            if expected_intermediate_events != -1:
                validate_intermediate_tensor()

    # Overwrite because unit test is failing, once passing, remove this function
    def test_qnn_backend_interpolate_nearest_2d(self):
        pass

    # Overwrite because unit test is failing, once passing, remove this function
    def test_qnn_backend_interpolate_bilinear_2d(self):
        pass
```

haowhsu-quic replied:
I see, some attributes are not added to `TestQNN` but only exist in `test_qnn_delegate.py`. Thank you for identifying this.

shewu-quic commented on backends/qualcomm/tests/utils.py:
```diff
@@ -197,6 +198,7 @@ def verify_output(  # noqa: C901
             )
 
             output_dir = f"{tmp_dir}/outputs"
+            os.makedirs(output_dir, exist_ok=True) 
```
How about using make_output_dir(output_dir) as #L231?

cccclai commented:
Can I get another round review on this?

haowhsu-quic commented on backends/qualcomm/tests/utils.py:
```diff
@@ -197,6 +198,7 @@ def verify_output(  # noqa: C901
             )
 
             output_dir = f"{tmp_dir}/outputs"
+            make_output_dir(output_dir) 
```
I think it's already there in L231.

haowhsu-quic replied:
Maybe you could help remove L231, thanks.

cccclai commented:
@haowhsu-quic is it good now?

haowhsu-quic commented:
> @haowhsu-quic is it good now?

Looks good, thank you!

cccclai committed:
Add buck file for unit test (#8106)

Summary:

```
buck2 test 'fbcode//mode/dev-nosan' fbcode//executorch/backends/qualcomm/tests/fb:test_qnn_delegate_simulator -- --exact 'executorch/backends/qualcomm/tests/fb:test_qnn_delegate_simulator - test_qnn_backend_mha (executorch.backends.qualcomm.tests.fb.test_qnn_delegate_simulator.TestQNNQuantizedOperatorSimulator)'
```

Differential Revision: D67463640
